sqflite/lib/src/sentry_batch.dart

```diff
@@ -0,0 +1,246 @@
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+import 'package:sqflite/sqflite.dart';
+
+// ignore: implementation_imports
+import 'package:sqflite_common/src/sql_builder.dart';
+
+import 'sentry_database.dart';
+
+/// A [Batch] wrapper that adds Sentry support.
+///
+/// ```dart
+/// import 'package:sqflite/sqflite.dart';
+/// import 'package:sentry_sqflite/sentry_sqflite.dart';
+///
+/// final database = await openDatabase('path/to/db');
+/// final sentryDatabase = SentryDatabase(database);
+/// final batch = sentryDatabase.batch();
+/// ```
+@experimental
+class SentryBatch implements Batch {
+  final Batch _batch;
+  final Hub _hub;
+
+  // we don't clear the buffer because SqfliteBatch don't either
+  final _buffer = StringBuffer();
+
+  /// ```dart
+  /// import 'package:sqflite/sqflite.dart';
+  /// import 'package:sentry_sqflite/sentry_sqflite.dart';
+  ///
+  /// final database = await openDatabase('path/to/db');
+  /// final sentryDatabase = SentryDatabase(database);
+  /// final batch = sentryDatabase.batch();
+  /// ```
+  SentryBatch(
+    this._batch, {
+    @internal Hub? hub,
+  }) : _hub = hub ?? HubAdapter();
+
+  @override
+  Future<List<Object?>> apply({bool? noResult, bool? continueOnError}) {
+    Future<List<Object?>> future() async {
+      final currentSpan = _hub.getSpan();
+
+      final span = currentSpan?.startChild(
+        SentryDatabase.dbSqlExecuteOp,
```

Comment:
`m`: can you execute SELECT queries in a batch? if so, I guess we should try to distinguish here between `db.sql.execute` and `db.sql.query` if possible? I think `sql.db.query` have special normalization on backend for N+1 detectors, so probably worth to look into that.

Reply:
Changing to `db` for now because of https://github.com/getsentry/sentry/blob/23e48d19c704d620a1345531512a61543ced0537/src/sentry/spans/grouping/strategy/base.py#L107
I'll ask if we can add the new ops to that file.

Reply:
https://github.com/getsentry/sentry/pull/45630